server/routes/components.ts

```diff
@@ -24,10 +25,47 @@ export default function routes({ serviceCatalogueService, redisService }: Servic
   })
 
   get('/veracode/data', async (req, res) => {
-    const components = await serviceCatalogueService.getComponents()
+    const resultFilters = getResultFilters(req.query.results as string)
+    const exemptionFilters = getExemptionFilters(req.query.exemption as string)
+    const allComponents = await serviceCatalogueService.getComponents(exemptionFilters)
+
+    const rows = allComponents
+      .filter(component => {
+        const passed = resultFilters.includes('passed')
+        const failed = resultFilters.includes('failed')
+        const unknown = resultFilters.includes('unknown')
+        const status = component.attributes.veracode_policy_rules_status
+
+        if ((passed && failed && unknown) || (!passed && !failed && !unknown)) {
+          return true
+        }
+
+        if (passed && !failed && !unknown && status === 'Pass') {
+          return true
+        }
+
+        if (passed && failed && !unknown && status !== null) {
+          return true
+        }
+
+        if (passed && !failed && unknown && (status === 'Pass' || status === null)) {
+          return true
+        }
+
+        if (!passed && failed && !unknown && status === 'Did Not Pass' && status !== null) {
```

Comment:
guess ` && status !== null` this is superfluous as you'd know it was `Did Not Pass`?

Reply:
Good call, I'll look at doing a bit of remedial work next to tidy this up.